service/routes/routes.js

```diff
@@ -0,0 +1,269 @@
+import { Deck } from '../models/Deck.js'
+import { User } from '../models/User.js'
+
+const isUrl = (value) => {
+    const re = /((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=\+\$,\w]+@)?[A-Za-z0-9.-]+|(?:www.|[-;:&=\+\$,\w]+@)[A-Za-z0-9.-]+)((?:\/[\+~%\/.\w-_]*)?\??(?:[-\+=&;%@.\w_]*)#?(?:[\w]*))?)/
+    return re.test(value)
+}
+
+const getCardById = async(req, res, next) => {
+    try {
+        const deck = await Deck.findOne({
+            'cards._id': req.params.id
+        })
+        if (deck) {
+            const card = deck.cards.find(card => card._id = req.params.id)
+            if (card) {
+                res.status(200).json(card)
+            } else {
+                res.status(404).send(`cannot find card ${req.params.id}`)
+            }
+        } else {
+            res.status(404).send(`cannot find deck associated with card ${req.params.id}`)
+        }
+
+    } catch (err) {
+        res.status(400).send(err.name + err.message)
+    }
+}
+
+const updateCardById = async(req, res, next) => {
+    const data = req.body
+    try {
+        const deck = await Deck.findOne({
+            'cards._id': req.params.id
+        })
+        if (deck) {
+            const card = deck.cards.find(card => card._id = req.params.id)
+            if (card) {
+                card.frontImage = data.frontImage ? data.frontImage : card.frontImage
```

Comment:
Good use of ternary operator. You can reduce this even further with the [Nullish Coalescing Operator, ??](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator)
```suggestion
                card.frontImage = data.frontImage ?? card.frontImage
```